src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -342,6 +342,17 @@ void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion
         break;
     }
 
+    // 2.B) Clamp the movement to the mutable viewport
+    const auto mutableViewport{ _GetMutableViewport() };
+    if (targetPos > mutableViewport.BottomRightInclusive())
+    {
+        targetPos = mutableViewport.BottomRightInclusive();
+    }
+    else if (targetPos < mutableViewport.Origin())

```

Comment:
won't this stop you from selecting above the *screen*? That seems very not correct :)